src/web/middleware/mod.rs

```diff
@@ -58,3 +62,42 @@ impl SyncServerRequest for HttpRequest {
         HawkIdentifier::extrude(self, method.as_str(), self.uri(), ci, secrets)
     }
 }
+
+pub fn emit_metric_for_4xx_error(
```

Comment:
The sentry middleware emits metrics similarly -- doing so for any error returning `Some` from `metric_label()` (maybe we should call it the `error_reporting` middleware instead).

Many of the `is_client_error` variants already have metric labels so they're already being reported. So I'd argue this isn't needed.

FYI we have a [dashboard panel started](https://earthangel-b40313e5.influxcloud.net/d/5kIPrfbWj/sync-rs-prod?orgId=1&refresh=1m&viewPanel=77) for showing these. It was only showing up a couple of the hawk variants but I've just updated it to show everything underneath "syncstorage.request." via a regex.

Reply:
Gotcha, I'll remove this then 👍🏻